contracts/chain-adapters/Linea_Adapter.sol

```diff
@@ -0,0 +1,354 @@
+     * @param _permitData The permit data for the token, if applicable.
+     */
+    function bridgeTokenWithPermit(
+        address _token,
+        uint256 _amount,
+        address _recipient,
+        bytes calldata _permitData
+    ) external payable;
+
+    /**
+     * @dev It can only be called from the Message Service. To finalize the bridging
+     *   process, a user or postmen needs to use the `claimMessage` function of the
+     *   Message Service to trigger the transaction.
+     * @param _nativeToken The address of the token on its native chain.
+     * @param _amount The amount of the token to be received.
+     * @param _recipient The address that will receive the tokens.
+     * @param _chainId The source chainId or target chaindId for this token
+     * @param _tokenMetadata Additional data used to deploy the bridged token if it
+     *   doesn't exist already.
+     */
+    function completeBridging(
+        address _nativeToken,
+        uint256 _amount,
+        address _recipient,
+        uint256 _chainId,
+        bytes calldata _tokenMetadata
+    ) external;
+
+    /**
+     * @dev Change the address of the Message Service.
+     * @param _messageService The address of the new Message Service.
+     */
+    function setMessageService(address _messageService) external;
+
+    /**
+     * @dev It can only be called from the Message Service. To change the status of
+     *   the native tokens to DEPLOYED meaning they have been deployed on the other chain
+     *   a user or postman needs to use the `claimMessage` function of the
+     *   Message Service to trigger the transaction.
+     * @param _nativeTokens The addresses of the native tokens.
+     */
+    function setDeployed(address[] memory _nativeTokens) external;
+
+    /**
+     * @dev Sets the address of the remote token bridge. Can only be called once.
+     * @param _remoteTokenBridge The address of the remote token bridge to be set.
+     */
+    function setRemoteTokenBridge(address _remoteTokenBridge) external;
+
+    /**
+     * @dev Removes a token from the reserved list.
+     * @param _token The address of the token to be removed from the reserved list.
+     */
+    function removeReserved(address _token) external;
+
+    /**
+     * @dev Linea can set a custom ERC20 contract for specific ERC20.
+     *   For security purpose, Linea can only call this function if the token has
+     *   not been bridged yet.
+     * @param _nativeToken address of the token on the source chain.
+     * @param _targetContract address of the custom contract.
+     */
+    function setCustomContract(address _nativeToken, address _targetContract) external;
+
+    /**
+     * @dev Pause the contract, can only be called by the owner.
+     */
+    function pause() external;
+
+    /**
+     * @dev Unpause the contract, can only be called by the owner.
+     */
+    function unpause() external;
+}
+
+interface IUSDCBridge {
```

Comment:
nit: for all of these external interfaces, I would remove all the functions (and events) that you don't call or use. For most of these, I think that would leave one function. I _would_ keep the natspec for the used functions, though.

Reply:
done here https://github.com/across-protocol/contracts-v2/pull/382/commits/4ae32a5a5bbc0e1175716dfa546bf22d03737e82